node-js-basics-101/node_modules/ws/lib/Receiver.js

```diff
@@ -0,0 +1,555 @@
+      if (!this.fin) {
+        this.error(new Error('FIN must be set'), 1002);
+        return;
+      }
+
+      if (compressed) {
+        this.error(new Error('RSV1 must be clear'), 1002);
+        return;
+      }
+
+      if (this.payloadLength > 0x7d) {
+        this.error(new Error('invalid payload length'), 1002);
+        return;
+      }
+    } else {
+      this.error(new Error(`invalid opcode: ${this.opcode}`), 1002);
+      return;
+    }
+
+    if (!this.fin && !this.fragmented) this.fragmented = this.opcode;
+
+    this.masked = (buf[1] & 0x80) === 0x80;
+
+    if (this.payloadLength === 126) this.state = GET_PAYLOAD_LENGTH_16;
+    else if (this.payloadLength === 127) this.state = GET_PAYLOAD_LENGTH_64;
+    else this.haveLength();
+  }
+
+  /**
+   * Gets extended payload length (7+16).
+   *
+   * @private
+   */
+  getPayloadLength16 () {
+    if (!this.hasBufferedBytes(2)) return;
+
+    this.payloadLength = this.readBuffer(2).readUInt16BE(0, true);
+    this.haveLength();
+  }
+
+  /**
+   * Gets extended payload length (7+64).
+   *
+   * @private
+   */
+  getPayloadLength64 () {
+    if (!this.hasBufferedBytes(8)) return;
+
+    const buf = this.readBuffer(8);
+    const num = buf.readUInt32BE(0, true);
+
+    //
+    // The maximum safe integer in JavaScript is 2^53 - 1. An error is returned
+    // if payload length is greater than this number.
+    //
+    if (num > Math.pow(2, 53 - 32) - 1) {
+      this.error(new Error('max payload size exceeded'), 1009);
+      return;
+    }
+
+    this.payloadLength = (num * Math.pow(2, 32)) + buf.readUInt32BE(4, true);
+    this.haveLength();
+  }
+
+  /**
+   * Payload length has been read.
+   *
+   * @private
+   */
+  haveLength () {
+    if (this.opcode < 0x08 && this.maxPayloadExceeded(this.payloadLength)) {
+      return;
+    }
+
+    if (this.masked) this.state = GET_MASK;
+    else this.state = GET_DATA;
+  }
+
+  /**
+   * Reads mask bytes.
+   *
+   * @private
+   */
+  getMask () {
+    if (!this.hasBufferedBytes(4)) return;
+
+    this.mask = this.readBuffer(4);
+    this.state = GET_DATA;
+  }
+
+  /**
+   * Reads data bytes.
+   *
+   * @private
+   */
+  getData () {
+    var data = constants.EMPTY_BUFFER;
+
+    if (this.payloadLength) {
+      if (!this.hasBufferedBytes(this.payloadLength)) return;
+
+      data = this.readBuffer(this.payloadLength);
+      if (this.masked) bufferUtil.unmask(data, this.mask);
+    }
+
+    if (this.opcode > 0x07) {
+      this.controlMessage(data);
+    } else if (this.compressed) {
+      this.state = INFLATING;
+      this.decompress(data);
+    } else if (this.pushFragment(data)) {
+      this.dataMessage();
+    }
+  }
+
+  /**
+   * Decompresses data.
+   *
+   * @param {Buffer} data Compressed data
+   * @private
+   */
+  decompress (data) {
+    const extension = this.extensions[PerMessageDeflate.extensionName];
+
+    extension.decompress(data, this.fin, (err, buf) => {
```

Comment:
__[jshint]__
* 'arrow function syntax (=>)' is only available in ES6 (use 'esversion: 6').

<!-- Comment by [SideCI](https://sideci.com) -->